language/compiler/ir-to-bytecode/syntax/src/lexer.rs

```diff
@@ -126,153 +126,167 @@ impl<'input> Lexer<'input> {
         self.previous_end
     }
 
+    pub fn lookahead(&self) -> Result<Tok, ParseError<usize, failure::Error>> {
+        let text = self.text.trim_start();
+        let whitespace = self.text.len() - text.len();
+        let start_offset = self.consumed + whitespace;
+        let (tok, _) = find_token(text, start_offset)?;
```

Comment:
This is probably fine for now, but if I understand this correctly, we don't save the result of the lookhead? So when it comes back to this token at peek it has to reperform the work?

Reply:
Yes, that's right. It's definitely something that we could optimize in the future.

Reply:
Cool, A job for another day though